### PR TITLE
Re-implement LM rescore for online transducer

### PR DESCRIPTION
**Pull request by SilverSulfide**
Shallow fusion can be too slow for online cpu inference. Added an option to use classical LM rescore instead.

- Rescore implementation based on https://github.com/k2-fsa/sherpa-onnx/pull/133
- Shallow fusion enabled by default
- Pass _lm-shallow-fusion=false_ to enable rescore instead
- Updated online Python API

CPU runtime comparison for ~1 min wav file using the default _--lm-num-threads=1_

| Method              | Runtime (s) | RTF |
| :---------------- | :------: | :----: |
| No LM | 5.7  | 0.098 |
| Rescore | 40 | 0.68 |
| Shallow Fusion | 69 | 1.2 |

### CODE REVIEW

File: sherpa-onnx/csrc/online-lm-config.cc [hunk 1]
```diff
@@ -34,7 +36,8 @@ std::string OnlineLMConfig::ToString() const {
 
   os << "OnlineLMConfig(";
   os << "model=\"" << model << "\", ";
-  os << "scale=" << scale << ")";
+  os << "scale=" << scale << ", ";
+  os << "shallow_fusion=\"" << (shallow_fusion ? "True" : "False") << "\")";
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/online-lm-config.cc]:
```suggestion
  os << "shallow_fusion=" << (shallow_fusion ? "True" : "False") << ")";
```

We use `""` only for string variables.

File: sherpa-onnx/python/sherpa_onnx/online_recognizer.py [hunk 2]
```diff
@@ -274,6 +275,7 @@ def from_transducer(
         lm_config = OnlineLMConfig(
             model=lm,
             scale=lm_scale,
```
Comment (csukuangfj) [inline sherpa-onnx/python/sherpa_onnx/online_recognizer.py:277]: Please also update
https://github.com/k2-fsa/sherpa-onnx/blob/5f9441bfdb2cc4d2a6076ac8be5e37a59f85f9fe/sherpa-onnx/python/csrc/online-lm-config.cc#L18

**Review by csukuangfj — CHANGES_REQUESTED** [review]

**Comment by SilverSulfide** [comment]
@csukuangfj hi, are there any more changes that should be written?

**Comment by csukuangfj** [comment]
Could you fix the CI errors for Python?

**Comment by SilverSulfide** [comment]
@csukuangfj should be resolved now!

**Pull request merged**